pkg/gui/controllers/commits_files_controller.go

```diff
@@ -153,6 +153,11 @@ func (self *CommitFilesController) checkout(node *filetree.CommitFileNode) error
 }
 
 func (self *CommitFilesController) discard(node *filetree.CommitFileNode) error {
+	parentContext, ok := self.c.CurrentContext().GetParentContext()
+	if !ok || parentContext.GetKey() != context.LOCAL_COMMITS_CONTEXT_KEY {
+		return self.c.ErrorMsg(self.c.Tr.CanOnlyDiscardFromLocalCommits)
+	}
```

Comment:
- Is this the right approach to solve it?
- Is it worth having integration tests for this?

Reply:
I'm happy with this fix. Ideally, we wouldn't have the keybinding there to start with but that would require either creating a separate view/context for files of a commit, or determining keybindings whenever a context gets focused (I actually think the first option makes more sense). But I'm fine with the approach this PR takes

Reply:
And yeah I think it's worth having an integration test for it

Reply:
Added a test in d7da6c7f5633.

Reply:
Hm, comparing this with #2707, we throw an error here but silently do nothing there. Should we change either of these to make the behavior consistent? If so, which one? Or is it ok the way it is?

Reply:
I'll leave a comment on 2707